portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,30 +56,63 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         if(comments.size() < numComments){
           long id = entity.getKey().getId();
           String text = (String) entity.getProperty("text");
-          long timestamp = (long) entity.getProperty("timestamp");
 
-          Comment comment = new Comment(id,text,timestamp);
+          // Do the translation.
+          Translate translate = TranslateOptions.getDefaultInstance().getService();
+          Translation translation =
+            translate.translate(text, Translate.TranslateOption.targetLanguage(language));
+          String translatedText = translation.getTranslatedText();
+          
+          long timestamp = (long) entity.getProperty("timestamp");
+          String email = (String) entity.getProperty("email");
+          String nickname = (String) entity.getProperty("nickname");
+          if(nickname == null || nickname == ""){
+              nickname = email;
+          }
+          Comment comment = new Comment(id,translatedText,timestamp,email, nickname);
           comments.add(comment);
         }
         else{
             break;
         } 
     }
     Gson gson = new Gson();
```

Comment:
You can store the Gson instance in a `private final` field in the class so you don't need to allocate a new instance for every request.